.travis.yml

```diff
@@ -1,40 +1,98 @@
 sudo: required
+_aliases:
+  - &ui-tests
+    addons:
+      firefox: latest-nightly
+      hosts: example.com
+    env: TOXENV=discopane-ui-tests DISPLAY=:99.0 GECKODRIVER=0.19.1
+    before_install:
+      - sudo apt-get -qq update
+      - sudo apt-get install jwm
+      - wget -O /tmp/geckodriver.tar.gz https://github.com/mozilla/geckodriver/releases/download/v$GECKODRIVER/geckodriver-v$GECKODRIVER-linux64.tar.gz
+      - mkdir $HOME/geckodriver && tar xvf /tmp/geckodriver.tar.gz -C $HOME/geckodriver
+      - export PATH=$HOME/geckodriver:$PATH
+      - firefox --version
+      - geckodriver --version
+      - pip install tox
+    before_script:
+      - yarn install
+      - sh -e /etc/init.d/xvfb start
+      - sleep 5
+      - jwm &
+      - sleep 5
+      - yarn start-func-test-server
+    script: tox
+  - &node6
+    addons:
+      firefox: latest
+    language: node_js
+    node_js: '6'
+    before_install: npm install -g yarn
+  - &node8
+    addons:
+      firefox: latest
+    language: node_js
+    node_js: '8'
+    before_install: npm install -g yarn
 services:
   - docker
-addons:
-  firefox: latest
-language: node_js
-node_js:
-- '6'
-- '8'
-env:
-  matrix:
-  # This is currently failing on master only.
-  # See https://github.com/mozilla/addons-frontend/issues/2691
-  # - TASK=build CHECK_BUNDLESIZE=1 ADDONS_FRONTEND_BUILD_ALL=1
-  - TASK=build ADDONS_FRONTEND_BUILD_ALL=1
-  - TASK=lint
-  - TASK=test-ci
-  - TASK=nsp-check
-  - TOXENV=discopane-ui-tests GUI=1 PYTEST_BASE_URL=http://localhost:4000
-  - TOXENV=flake8
-  - TOXENV=dennis-lint
 cache:
   yarn: true
   directories:
   - node_modules
-before_install:
-- npm install -g yarn
-- if [ $TOXENV ]; then sudo pip install tox; fi
-- if [[ $TOXENV == 'discopane-ui-tests' ]]; then wget -O geckodriver.tar.gz https://github.com/mozilla/geckodriver/releases/download/v0.14.0/geckodriver-v0.14.0-linux64.tar.gz; fi
-- if [[ $TOXENV == 'discopane-ui-tests' ]]; then gunzip -c geckodriver.tar.gz | tar xopf -; fi
-- if [[ $TOXENV == 'discopane-ui-tests' ]]; then chmod +x geckodriver && sudo mv geckodriver /usr/local/bin; fi
-before_script:
-- if [ $GUI ]; then export DISPLAY=:99.0 && sh -e /etc/init.d/xvfb start && sleep 3; fi
-script:
-- if [[ $TOXENV == 'discopane-ui-tests' ]]; then
-    yarn start-func-test-server;
-  fi
-- if [ $TOXENV ]; then tox; fi
-- if [ $TASK ]; then yarn $TASK; fi
-- if [ $CHECK_BUNDLESIZE ]; then yarn bundlesize; fi
+jobs:
+  include:
+    - stage:
+      language: python
+      python: 3.6
+      <<: *ui-tests
+      install: nvm install 6
+    - stage:
+      language: python
+      python: 3.6
+      <<: *ui-tests
+      install: nvm install 8
```

Comment:
I don't like how there are always two duplicate stages, one for each version of Node. The old approach to this yaml file using `matrix` did not need that. Is there a way to switch back to `matrix` but still use aliases to share configuration?

Reply:
Well its the same, it just look different. The jobs config needs to be more explicitly written whereas the matrix would create all of it.